### PR TITLE
Minecraft server

### PR DESCRIPTION
**Pull request by jaxsnjohnson**

### CODE REVIEW

File: content/Minecraft-Server-Hosting.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+  - Minecraft
+  - Mojang
+  - Games
+hidden: false
+section: software
+---
+
+# Hosting A Minecraft Server
+
+Minecraft 1.17 dropped on June 8, 2021. In this article, we will be going over how to set up a Minecraft server on your Pop!_OS or Ubuntu install. A Minecraft System76 recommends having a minimum of 8Gb of RAM but for a good experience, 16Gb of RAM is recommended. Any hard drive over 128GB for a basic server but if you want to backups 1TB to 2TB is recommended. This article will live mostly in the terminal, if you want a refresher on the terminal you can see our article [here](https://support.system76.com/articles/terminal-basics).
```
Comment (storyaddict) [inline content/Minecraft-Server-Hosting.md:16]: Typo: "A Minecraft System76 recommends"

Should be: "System76 OR Mojang recommends..."

Comment (storyaddict) [reply content/Minecraft-Server-Hosting.md:16]: 8GB minimum.

16GB "recommended" or "preferred."

Comment (storyaddict) [reply content/Minecraft-Server-Hosting.md:16]: Any hard drive 128GB for a basic server... <- sentence fragment.

Comment (storyaddict) [reply content/Minecraft-Server-Hosting.md:16]: This article will use mostly Terminal commands.

File: content/Minecraft-Server-Hosting.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+## Updating and Installing Software Needed
+
+To install the Minecraft server we will need to install Java (listed as default-jdk), wget, screen, and nmap.  Java is what Minecraft runs in and is required for the server to run - we are also marking it as hold so no future updates are applied (some java updates can break the server). wget will allow us to download the Minecrafter Server from a URL hosted by Mojang, this way we can put the file in the directory directly instead of using a web browser.  screen will allow us to run commands and the server in an environment we do not need to have open or be connected to - think of it as a virtual terminal instance you can connect and disconnect from.  nmap is a network scanner that we will use for testing and network configuration.
```
Comment (storyaddict) [inline content/Minecraft-Server-Hosting.md:20]: Redundant wording: "Java is what Minecraft **runs** in and is required for the server to **run**"

Comment (storyaddict) [reply content/Minecraft-Server-Hosting.md:20]: marking the java package as "held." or putting the java package on hold.

Comment (storyaddict) [reply content/Minecraft-Server-Hosting.md:20]: comma splice: 

Server from a URL hosted by Mojang[.] Downloading Minecraft Server in this way means we can...

Comment (storyaddict) [reply content/Minecraft-Server-Hosting.md:20]: . Screen will allow...

I think there's a clearer way of wording this. The virtual terminal explanation is a good explanation.

Comment (storyaddict) [reply content/Minecraft-Server-Hosting.md:20]: OR `screen` and `nmap` to show them as Terminal name schemes.

File: content/Minecraft-Server-Hosting.md [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+sudo apt clean  
+sudo apt update -m  
+sudo dpkg --configure -a  
+sudo apt install -f  
+sudo apt full-upgrade -y
+sudo apt install wget screen default-jdk nmap -y
+sudo apt-mark hold default-jdk
+```
+
+*Please note not all of these commands will output something.*
```
Comment (storyaddict) [inline content/Minecraft-Server-Hosting.md:34]: > **NOTE:** not all these commands will produce output.

File: content/Minecraft-Server-Hosting.md [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+
+## Make The Mincraft Group & User & Minecraft Direcory
+The Minecraft server will make a lot of files for the world, settings, plugins, and other resources.  The following commands are to make the directory for the servers to exist in.  We will want to make a group and a user for the server to operate in and use.  We will need to then make a directory file to live in and make sure that the new server user has ownership of the directory.
```
Comment (storyaddict) [inline content/Minecraft-Server-Hosting.md:37]: The following commands are to make the directory for the servers to exist in.

CHANGE TO:

 The following commands make the directories for the servers.

File: content/Minecraft-Server-Hosting.md [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+sudo chown -Rv mchost /mchost/~
+```
+## Installing Minecaft Server
+
+```bash
+sudo wget -O /mchost/v-1-17/live/minecraft_server.jar https://launcher.mojang.com/v1/objects/0a269b5f2c5b93b1712d0f5dc43b6182b9ab254e/server.jar
+sudo bash -c "echo eula=true > /mchost/v-1-17/live/eula.txt"
+sudo chown -R mchost /mchost/
+```
+*Please note the server Jar will change over time and you will want to go to Mojan's website to grab the newest Jar.*
```
Comment (storyaddict) [inline content/Minecraft-Server-Hosting.md:54]: Formatting:

> **NOTE:** the server Jar...

File: content/Minecraft-Server-Hosting.md [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+su mchost
+java -Xms4G -Xmx4G -XX:+UseG1GC -XX:+UnlockExperimentalVMOptions -XX:MaxGCPauseMillis=100 -XX:+DisableExplicitGC -XX:TargetSurvivorRatio=90 -XX:G1NewSizePercent=50 -XX:G1MaxNewSizePercent=80 -XX:G1MixedGCLiveThresholdPercent=50 -XX:+AlwaysPreTouch -jar server.jar
+```
+
+```
+The Lower Bound: -Xms4G
+The Upper Bound: -Xmx4G
+```
+
+These values can be changed up to around 10GB if you need to go over 10GB.  This rarely needs to be done on smaller servers unless you have complex data packs, a large world file, or a loss of players (around 100+).  If you need these values to be higher please change the G1 Max and New Size Percent to the following:
```
Comment (storyaddict) [inline content/Minecraft-Server-Hosting.md:69]: *or lots of players

**Review by storyaddict — CHANGES_REQUESTED** [review]
See in-line comments.

File: content/Minecraft-Server-Hosting.md (hunk 1 above)
Comment (jaxsnjohnson) [reply content/Minecraft-Server-Hosting.md]: Corrected spelling/grammar

File: content/Minecraft-Server-Hosting.md (hunk 2 above)
Comment (jaxsnjohnson) [reply content/Minecraft-Server-Hosting.md]: corrected spelling and grammar

**Review by gangwerz — CHANGES_REQUESTED** [review]
I am not seeing anything wrong with the article. However, the new Markdown to HTML compiler/transpiler has not been rendering in-line code (single back ticks), so any of those sections may need to be rewritten.

**Comment by jaxsnjohnson** [comment]
> I am not seeing anything wrong with the article. However, the new Markdown to HTML compiler/transpiler has not been rendering in-line code (single back ticks), so any of those sections may need to be rewritten.

Should I just make it bold or do we have an option for an equivalent feature?

**Comment by gangwerz** [comment]
> > I am not seeing anything wrong with the article. However, the new Markdown to HTML compiler/transpiler has not been rendering in-line code (single back ticks), so any of those sections may need to be rewritten.
> 
> Should I just make it bold or do we have an option for an equivalent feature?

I think we should share with the team, cause it may require updating most articles, or another update to the markdown processor
Example on the live site: https://support.system76.com/articles/terminal-basics/

**Pull request merged**